### PR TITLE
Add filter for GreenLanes::Categorisation.

### PR DESCRIPTION
**Pull request by alexdesi**
### Jira link
https://transformuk.atlassian.net/browse/HOTT-4351

### What?
Add filter for GreenLanes::Categorisation.

### CODE REVIEW

File: app/models/green_lanes/categorisation.rb [hunk 1]
```diff
@@ -31,6 +31,18 @@ def load_from_string(data)
       def all
         @all ||= load_from_file
       end
+
+      def filter(regulation_id:, measure_type_id:, geographical_area: nil)
+        return [] if regulation_id.blank? || measure_type_id.blank?
+
+        all.select { |c| c.match?(regulation_id:, measure_type_id:, geographical_area:) }
```
Comment (jebw) [inline app/models/green_lanes/categorisation.rb:38]: super minor suggestion but you could just use `...` to pass all params through?

File: spec/models/green_lanes/categorisation_spec.rb [hunk 2]
```diff
@@ -101,4 +101,46 @@
...
+      subject(:categorisation_filter) do
+        described_class.filter regulation_id: 'D000004', measure_type_id: '430', geographical_area: 'US'
+      end
+
+      it { expect(categorisation_filter.first).to have_attributes regulation_id: 'D000004', measure_type_id: '430' }
+    end
+
+    context 'when geographical_area does not match' do
+      subject(:categorisation_filter) do
+        described_class.filter regulation_id: 'D000004', measure_type_id: '430', geographical_area: 'unexisting'
```
Comment (jebw) [inline spec/models/green_lanes/categorisation_spec.rb:139]: how about `non-existant` ? `unexisting` isn't really a word AFAIK?

File: spec/models/green_lanes/categorisation_spec.rb [hunk 3]
```diff
@@ -101,4 +101,46 @@
...
+      end
+
+      it { is_expected.to be_an Array }
```
Comment (jebw) [inline spec/models/green_lanes/categorisation_spec.rb:142]: TIL - `be_an` :smile:

**Review by jebw — APPROVED** [review]
LGTM - couple of really minor suggestions

File: app/models/green_lanes/categorisation.rb (hunk 1 above)
Comment (alexdesi) [reply app/models/green_lanes/categorisation.rb:38]: in this case I'd prefer to be more explicit

File: spec/models/green_lanes/categorisation_spec.rb (hunk 2 above)
Comment (alexdesi) [reply spec/models/green_lanes/categorisation_spec.rb:139]: good point, I'll chat it.
Strangely some dictionaries still report it as a proper word, but they consider "non-existent"   wrong now.

**Review by jebw — APPROVED** [review]
Nice, super clean!

**Pull request merged**